### PR TITLE
RobotScript optimisations

### PR DESCRIPTION
**Pull request by Hexicube**
Removed almost all instances of transform.Find(x)

### CODE REVIEW

File: UnityProject/Assets/Scripts/RobotScript.cs [hunk 1]
```diff
@@ -332,12 +332,11 @@ class HoleAttachment {
...
     			if(gun_delay <= 0.0f){
     				gun_delay += 0.1f;
-    				Transform point_muzzle_flash = gun_pivot.Find("gun").Find("point_muzzleflash");
-    				Instantiate(muzzle_flash, point_muzzle_flash.position, point_muzzle_flash.rotation);
+    				Instantiate(muzzle_flash, muzflash_point.position, muzflash_point.rotation);
     				PlaySoundFromGroup(sound_gunshot, 1.0f);
     				
-    				GameObject bullet = (GameObject)Instantiate(bullet_obj, point_muzzle_flash.position, point_muzzle_flash.rotation);
-    				bullet.GetComponent<BulletScript>().SetVelocity(point_muzzle_flash.forward * 300.0f);
+    				GameObject bullet = (GameObject)Instantiate(bullet_obj, muzflash_point.position, muzflash_point.rotation);
```
Comment (Kroporo) [inline UnityProject/Assets/Scripts/RobotScript.cs]: typo. should be called "point_muzzle_flash"

**Comment by autious** [comment]
Can you verify that it compiles and runs @Kroporo?

**Comment by Hexicube** [comment]
Renamed, though it may not compile now. The gun turret will need the muzzle flash empty re-linked in the editor, I can't currently do that.

**Comment by Hexicube** [comment]
Manually renamed the links, it should work fine.

File: UnityProject/Assets/Scripts/RobotScript.cs [hunk 2]
```diff
@@ -23,6 +23,8 @@ public class RobotScript:MonoBehaviour{
     public AudioClip sound_engine_loop;
     public AudioClip sound_damaged_engine_loop;
 
+    public Transform point_pivot, top_rotor, bottom_rotor, point_spark, camera_pivot, battery, gun_pivot, point_muzzle_flash, gun_camera, drone_camera, motor;
```
Comment (Kroporo) [inline UnityProject/Assets/Scripts/RobotScript.cs]: Don't think this shortcut has been used anywhere else in Receiver's code,

Comment (autious) [reply UnityProject/Assets/Scripts/RobotScript.cs]: I agree, probably looks better if each transform is declared on a separate line for consistency.

**Pull request merged**